SIL.Media/MediaInfo.cs

```diff
@@ -1,300 +1,260 @@
-			{
-				var regex = new Regex(", ([^,]*) channels");
-				if (regex.IsMatch(ffmpegOutput))
-				{
-					var match = regex.Match(ffmpegOutput);
-					var value = match.Groups[1].Value;
-
-					int channelCount;
+		public IMediaAnalysis AnalysisData { get; }
 
-					if (int.TryParse(value, out channelCount))
-					{
-						this.ChannelCount = channelCount;
-					}
-					return;
-				}
-				// older versions (i.e. 9.18 which comes with trusty) have a slightly different
-				// output format
-				regex = new Regex(", (mono|stereo|quad|hexagonal|octagonal|downmix|[2-7].[0-1](\\(.\\))?),");
-				if (regex.IsMatch(ffmpegOutput))
-				{
-					var match = regex.Match(ffmpegOutput);
-					var value = match.Groups[1].Value;
-					switch (value)
-					{
-						case "mono":
-							this.ChannelCount = 1;
-							break;
-						case "stereo":
-						case "downmix":
-							this.ChannelCount = 2;
-							break;
-						case "2.1":
-						case "3.0":
-						case "3.0(back)":
-							this.ChannelCount = 3;
-							break;
-						case "3.1":
-						case "4.0":
-						case "quad":
-						case "quad(side)":
-							this.ChannelCount = 4;
-							break;
-						case "4.1":
-						case "5.0":
-							this.ChannelCount = 5;
-							break;
-						case "5.1":
-						case "6.0":
-						case "6.0(front)":
-						case "hexagonal":
-							this.ChannelCount = 6;
-							break;
-						case "6.1":
-						case "6.1(front)":
-						case "7.0":
-						case "7.0(front)":
-							this.ChannelCount = 7;
-							break;
-						case "7.1":
-						case "7.1(wide)":
-						case "octagonal":
-							this.ChannelCount = 8;
-							break;
-					}
-				}
-			}
-
-			private void ExtractSamplesPerSecond(string ffmpegOutput)
+		private MediaInfo(IMediaAnalysis mediaAnalysis)
+		{
+			AnalysisData = mediaAnalysis;
+			if (mediaAnalysis != null)
 			{
-				var match = Regex.Match(ffmpegOutput, ", ([^,]*) Hz");
-				if (match.Groups.Count == 2)
-				{
-					var value = match.Groups[1].Value;
-
-					int frequency;
-
-					if (int.TryParse(value, out frequency))
-					{
-						this.SamplesPerSecond = frequency;
-					}
-				}
+				if (mediaAnalysis.PrimaryAudioStream != null)
+					Audio = new AudioInfo(mediaAnalysis);
+				if (mediaAnalysis.PrimaryVideoStream != null)
+					Video = new VideoInfo(mediaAnalysis);
 			}
+		}
 
-			private void ExtractBitDepth(string ffmpegOutput)
-			{
-
-				var match = Regex.Match(ffmpegOutput, @"pcm_s(\d+)");
-				if (match.Groups.Count == 2)
-				{
-					var value = match.Groups[1].Value;
-
-					//todo: find out if we will really get s16, s24, s96, etc.
-					int depth;
-					if (int.TryParse(value, out depth))
-					{
-						BitDepth = depth;
-						return;
-					}
-				}
-				//no pcm found, use the myserious "s24, s32" stuff, which may be misleading
-
-				 match = Regex.Match(ffmpegOutput, @", s(\d+),");
-				if (match.Groups.Count == 2)
-				{
-					var value = match.Groups[1].Value;
+		public static string MissingComponentMessage => Format(
```

Comment:
```suggestion
		[PublicAPI]
		public static string MissingComponentMessage => Format(
```

(and more places below)